public/app/plugins/panel/graph/module.ts

```diff
@@ -351,4 +354,81 @@ class GraphCtrl extends MetricsPanelCtrl {
   };
 }
 
-export { GraphCtrl, GraphCtrl as PanelCtrl };
+// Use new react style configuration
+export const plugin = new PanelPlugin<GraphPanelOptions, GraphFieldConfig>(null)
+  .useFieldConfig({})
+  .setPanelOptions(builder => {
+    builder
+      .addBooleanSwitch({
+        category: 'Draw Mode',
+        path: 'bars',
+        name: 'Bars',
+        description: 'show bars',
+        defaultValue: false,
+      })
+      .addBooleanSwitch({
+        category: 'Draw Mode',
+        path: 'lines',
+        name: 'Lines',
+        description: 'show bars',
+        defaultValue: true,
+      })
+      .addBooleanSwitch({
+        category: 'Draw Mode',
+        path: 'points',
+        name: 'Points',
+        description: 'show bars',
+        defaultValue: false,
+      })
+      .addBooleanSwitch({
+        category: 'Legend',
+        path: 'legend.isVisible',
```

Comment:
@dprokop -- are nested values supported?  or does this have to be all at the root level?

Reply:
yes, this should be supported

Reply:
yes they are!